pkg/actuators/machine/stubs.go

```diff
@@ -32,32 +36,90 @@ const (
 	stubKeyName                 = "alibabacloud-actuator-key-name"
 	stubClusterID               = "alibabacloud-actuator-cluster"
 	stubImageID                 = "centos_7_9_x64_20G_alibase_20210318.vhd"
-	stubVpcID                   = "vpc-3ze4u29pd4lniym7i1xnp"
-	stubVSwitchID               = "vsw-7ze567qrl5das7q8s4rei"
-	stubInstanceID              = "i-2ze3hj0qh9d290rpax7w"
-	stubSecurityGroupID         = "sg-2zeebk9qd965vc2xqq4w"
+	stubVpcID                   = "vpc-vk6f1qfd3w77gnmh"
+	stubVSwitchID               = "vsw-sc0w64w2s3d9s8cu"
+	stubInstanceID              = "i-bg2ss7v5ck5skyp9"
+	stubSecurityGroupID         = "sg-h8ympu5av8hhtwks"
+	stubResourceGroupID         = "rg-6ljxzbpksxaa0buw"
+	stubResourceGroupName       = "test-rg"
 	stubSystemDiskCategory      = "cloud_essd"
 	stubSystemDiskSize          = 120
 	stubInternetMaxBandwidthOut = 100
 	stubPassword                = "Hello$1234"
 	stubInstanceType            = "ecs.c6.2xlarge"
+	stubInstanceStatus          = "Running"
+
+	stubRunningInstanceStauts  = ECSInstanceStatusRunning
+	stubPendingInstanceStatus  = ECSInstanceStatusPending
+	stubStartingInstanceStatus = ECSInstanceStatusStarting
+	stubStoppingInstanceStatus = ECSInstanceStatusStopping
+	stubStoppedInstanceStatus  = ECSInstanceStatusStopped
+	stubReleasedInstanceStatus = "Released"
 )
 
-func stubAlibabaCloudCredentialsSecret() *corev1.Secret {
+const userDataBlob = `#!/bin/bash
+echo "test"
+`
+
+func stubUserDataSecret() *corev1.Secret {
 	return &corev1.Secret{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      alibabaCloudCredentialsSecretName,
+			Name:      alibabaCloudMasterUserDataSecretName,
 			Namespace: defaultNamespace,
 		},
+		Data: map[string][]byte{
+			userDataSecretKey: []byte(userDataBlob),
+		},
+	}
+}
+
+func stubAlibabaCloudCredentialsSecret() *corev1.Secret {
+	return GenerateAlibabaCloudCredentialsSecretFromEnv(alibabaCloudCredentialsSecretName, defaultNamespace)
+}
+
+func GenerateAlibabaCloudCredentialsSecretFromEnv(secretName, namespace string) *corev1.Secret {
```

Comment:
What's the reason for making this public? Is it being used in other test suites?

Reply:
change to generateAlibabaCloudCredentialsSecretFromEnv